packages/react-hooks/src/useSetTimeout.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { useConst } from './useConst';
+import { useEffect } from 'react';
+
+export type UseSetTimeoutReturnType = {
+  setTimeout: (callback: () => void, duration: number) => number;
+  clearTimeout: (id: number) => void;
+};
+
+/**
+ *  Returns a wrapper function for `setTimeout` which automatically handles disposal.
+ */
+export const useSetTimeout = (): UseSetTimeoutReturnType => {
```

Comment:
My only suggestion would be that the useSetTimeout and useSetInterval are almost identical; it seems you could abstract these into one function:
```tsx
const createUseTimer = (set, clear) => { ... everything above };

export const useSetTimeout = createUseTimer(setTimeout, clearTimeout);
export const useSetInterval = createUseTimeout(setInterval, clearInterval);
```

This would reduce the overall bundle hit and redundant code.

Reply:
I like this too, though I'd call it `useTimers` instead.

Reply:
Oh never mind I see what you mean. I think I actually prefer to leave it as-is because that feels like a confusing amount of indirection, and the bundle hit and amount of redundant code is pretty small.

Reply:
The whole block is duplicate though. We should try and right DRY code if possible :) Perhaps comments would help to clarify. I am going to guess it would shave off another 500 bytes, also keeps the code condensed so that fixing or optimizing it is done one time. Every bit counts.

Reply:
On second thought, keeping them separate is probably cleaner and not worth the savings. The API surface returned is different and it would be less readable.